clickhouse/CMakeLists.txt

```diff
@@ -110,15 +118,23 @@ SET_TARGET_PROPERTIES (clickhouse-cpp-lib
     VERSION ${CLICKHOUSE_CPP_VERSION}
 )
 TARGET_LINK_LIBRARIES (clickhouse-cpp-lib
-    absl::int128
```

Comment:
Why did you mark dependencies private?

Reply:
Since all dependencies are static libraries the exported target should not propagate these dependencies further downstream.

Reply:
Clickhouse-cpp-lib can be built as shared library

Reply:
Yes, this is how I build it for packaging. The dependencies are still static though.

Reply:
> Yes, this is how I build it for packaging. The dependencies are still static though.

Dependencies may be shared, for example, conan packaging. In this case, looks like package will be broken.

Reply:
The package should not be broken, but the exported CMake target config will be missing dependencies. Is clickhouse-cpp built with conan? I don't see any configuration for it anywhere.

Reply:
https://conan.io/center/recipes/clickhouse-cpp?version=2.5.1 and here is an example of usage https://github.com/userver-framework/userver/blob/develop/conanfile.py#L162

Reply:
Okay, it seems like conan generates its own CMake config file, so shouldn't be affected.
https://github.com/conan-io/conan-center-index/blob/master/recipes/clickhouse-cpp/all/conanfile.py#L106

I know nothing about conan though, so I'd be grateful if you verify this.

Reply:
> it seems like conan generates its own CMake config file, so shouldn't be affected
Okay, I forgot about this. From this point of view, everything is ok. Thank you :)

Reply:
There is of course some room for improvement in how I generate the config file (it's probably not really how you are supposed to do it) and it will most likely not cover all possible usecases. I figured out that it might still be useful as is, after all it's an improvement from not having one at all.